packages/compass/src/app/intercom.ts

```diff
@@ -0,0 +1,200 @@
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          x.parentNode!.insertBefore(s, x);
+          debug('intercom script injected');
+          w.Intercom('boot', metadata);
+          debug('intercom booted with metadata', metadata);
+          resolve();
+        } catch (e) {
+          reject(e);
+        }
+      }
+      if (d.readyState === 'complete') {
+        // call directly, window already loaded
+        l();
+      } else if (w.attachEvent) {
+        w.attachEvent('onload', l);
+      } else {
+        w.addEventListener('load', l, false);
+      }
+    }
+  }).catch((e) => {
+    debug('failed to inject intercom script, reason:', e);
+  });
+}
+
+export function unloadIntercom(): void {
+  try {
+    const win: any = window;
+
+    debug('unload intercom');
+    if (typeof window === 'undefined') {
+      return;
+    }
+
+    // remove the intercom widget
+    if (win.Intercom) {
+      win.Intercom('hide');
+      win.Intercom('shutdown');
+    }
+
+    const el = document.getElementById(INTERCOM_SCRIPT_ELEM_ID);
+    if (el) {
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      el.parentNode!.removeChild(el);
+    }
+    debug('intercom script unloaded');
+  } catch (e) {
+    debug('unload intercom failed, reason:', e);
+  }
+}
+
+export async function setupIntercom(
+  preferences: Pick<EventEmitter, 'on'> & {
+    isFeatureEnabled: (feature: string) => boolean;
+  },
+  user: {
+    id: string;
+    createdAt: Date;
+  }
+): Promise<void> {
+  if (typeof window === 'undefined') {
+    return;
+  }
+
+  if (process.env.HADRON_ISOLATED === 'true') {
+    debug('Skipping intercom setup on HADRON_ISOLATED');
+    return;
+  }
+
+  const intercomAppId = process.env.HADRON_METRICS_INTERCOM_APP_ID;
+
+  if (!intercomAppId) {
+    debug(
+      'skipping Intercom setup HADRON_METRICS_INTERCOM_APP_ID is not defined'
+    );
+    return;
+  }
+
+  const metadata: IntercomMetadata = {
+    user_id: user.id,
+    app_id: intercomAppId,
+    created_at: Math.floor(user.createdAt.getTime() / 1000),
+    app_name: process.env.HADRON_PRODUCT_NAME,
+    app_version: process.env.HADRON_APP_VERSION,
+    app_stage: process.env.NODE_ENV,
+  };
+
+  ('MongoDB Compass Dev Local');
+
+  const intercomWidgetUrl =
+    'https://widget.intercom.io/widget/' + intercomAppId;
+
+  // in some environment the network can be firewalled, this is a safeguard to avoid
+  // uncaught errors when injecting the script.
+
+  debug('testing intercom availability');
+  const response = await fetch(intercomWidgetUrl);
+  if (response.status >= 400) {
+    debug('intercom unreachable, skipping setup');
+    return;
+  }
+
+  debug('intercom is reachable, proceeding with the setup');
+
+  // we use a queue to ensure that the state of intercom will eventually reflect
+  // what is selected in the preferences.
+  const queue = new PQueue({ concurrency: 1 });
```

Comment:
Feel free to ignore, but fwiw, this seems like something that’s similarly easy to do without an added external dependency:

```js
const queue = Promise.resolve();

...

queue = queue
  .then(() => loadIntercomScript(intercomWidgetUrl, metadata))
  .catch((e) => debug('queue error', e));
```

Reply:
makes a lot of sense, much better without the dep